Indexer/Indexer.php

```diff
@@ -172,7 +166,7 @@ public function discoverEntity($entity_or_class, EntityManager $em)
         // check if we already saw this type of entity
         // to avoid some expensive work
         if (!array_key_exists($class, self::$indexSettings)) {
-            self::$indexSettings[$class] = $this->getMetaDataLoader()->getMetaData($entity, $em);
+            self::$indexSettings[$class] = $this->loaderChain->getMetaData($entity, $em);
         }
 
         return false !== self::$indexSettings[$class];
```

Comment:
Right now, `self::$indexSettings[$class]` is `null` if the metadata has not been loaded yet, but `false` if it has been loaded but the class in question is not indexed. `discoverEntity` thus returns `true` if the class should be indexed, `false` if it shouldn't. However, with the new loaders, this functionality won't work since yaml loader returns `null` when it can't load metadata, so `discoverEntity` will always return `true`, even when there's no metadata present. I hope this explains it better.